README.md

```diff
@@ -51,24 +55,46 @@ Set<String> usernames = PreferenceHelper.get("all_names", null, Set.class);
 ```
 Note that primitive types (int, boolean, float, long) cannot be null.
 
-
+### Contains
 You can also check if a value has been stored with: 
 ```
 boolean key_exists = PreferenceHelper.contains("some_key");
 ```
 
+### Removing
+You can remove an object at a key with:
+```
+PreferenceHelper.remove("some_key");
+```
+
+### Custom Objects
 The PreferenceHelper can also store non-primitive objects.  Your object will be serialized and stored as a JSON string.
 ```
 PreferenceHelper.put("current_user", user);
 User user = PreferenceHelper.get("current_user", null, User.class);
 ```
+### Custom Objects with Type Parameters
+**NOTE: As of now, the library is not able to easily retrieve objects that use type parameter aside from List**
+
+Due to type erasure at compile time, the library will not know what type parameter type to convert the objects to.
+For example, retrieving something like `Response<APIResponse>` likely will not work properly as it will return you a generic Response object with no type param set.
+**However, a workaround for retrieving a List of objects has been added...**
+```
+List<User> fallbackList = new ArrayList<>();
+List<User> userList = PreferenceHelper.getList("list_key", fallbackList, User[].class);
+```
+This will look up the object array stored at that key and turn it into a List of that object type.
+If nothing is stored, the fallback will be returned.  If a null value is stored, a null list will be returned.
+If the stored object at the specified key is not a valid array or list of the desired object, an exception will be thrown.
+
+**NOTE: This only applies to retrieving objects as a substitute for `get()`. You should still use `put()` to store a List.**
```

Comment:
Remove this and reference `putList()`

Reply:
I did this but I end up kind of rewriting most of the lines in the last commit.